### DISCUSSION TITLE
Switching to the production mode of Keycloak, or how to remove "Running the server in development mode. DO NOT use this configuration in production."

### DISCUSSION THREAD

**Discussion by mathieucaroff** [keycloak/keycloak#35445] [Q&A]
Whenever I start keycloak in docker, I get the below message displayed in the docker logs. I've looked into the source code of Keycloak and Quarkus; I've tried:

- setting the environement variable `KC_PROFILE` to `prod` in docker-compose as well as in the Dockerfile
- passing `--profile=prod` in the ENTRYPOINT or CMD fields of the Docker image
- passing `start` in the ENTRYPOINT or CMD fields of the Docker image
- I've tried Keycloak images 24.0.5 and 26.0.6

Whatever I do, I always get the below message in the logs of Keycloak:

```txt
[org.keycloak.quarkus.runtime.KeycloakMain] (main) Running the server in development mode. DO NOT use this configuration in production.
```

The message appears after about 35 seconds of starting the container.

What am I supposed to do to switch Keycloak to production mode?

**Comment by arichtman-srt** [comment]
I think we use `KC_START_CMD` but the docs say just pass `start` as your command [Edit] and Bitnami do that [here](https://github.com/bitnami/containers/blob/73f15a1622ae265f24449fae8d387d55123dc2fc/bitnami/keycloak/26/debian-12/rootfs/opt/bitnami/scripts/keycloak/run.sh#L24)

Edit: `KC_START_CMD` seems to be an in-house thing.

https://www.keycloak.org/server/containers#_starting_the_optimized_keycloak_container_image
